src/parser/permit-generation-module.ts

```diff
@@ -53,6 +54,11 @@ export class PermitGenerationModule implements Module {
       console.warn("[PermitGenerationModule] Invalid env detected, skipping.");
       return Promise.resolve(result);
     }
+    const isPrivateKeyAllowed = await this._isPrivateKeyAllowed(payload.evmPrivateEncrypted, program.eventPayload.organization?.id || 0, program.eventPayload.repository.id);
```

Comment:
I decided that we should rely on the repository ID first, and use the organization ID as a fallback in case no config is defined for some reason. 

The ubiquibot-config repository is intended to be private, which means that no external contributors can guess the ID, making it a suitable salt. 

Anybody can query anybody's organization ID, and create a malicious "drainer" plugin though (after obtaining their evm private key encrypted of course)

Reply:
> use the organization ID as a fallback in case no config is defined for some reason.

If config is not defined then `conversation-rewards` plugin simply won't run.

> The ubiquibot-config repository is intended to be private, which means that no external contributors can guess the ID, making it a suitable salt.

Repository id is not a suitable salt.
```
rndquu@laptop ~ % gh repo view ubiquity/ubiquity-dollar --json id
{
  "id": "MDEwOlJlcG9zaXRvcnkzOTQ3Nzc4NjI="
}
```

Where repository id `MDEwOlJlcG9zaXRvcnkzOTQ3Nzc4NjI=` is `base64` representation of:
```
010:Repository394777862
```

Where `394777862` is `29.9` bits of entropy:
```
Length: 9
Possible symbols: 10
Possible combinations: 10^9 = 1000000000
Bits of entropy: log (base 2) 1000000000 = 29.9
```

> Anybody can query anybody's organization ID, and create a malicious "drainer" plugin though (after obtaining their evm private key encrypted of course)

No. Malicious "drainer" will have to ask a partner to encrypt partner's wallet private key with the drainer's `X25519_PUBLIC_KEY` which the "drainer" plugin would then decrypt using a malicious `X25519_PRIVATE_KEY`. In reality, even if a "drainer" plugin uses a partner's encrypted private key (i.e. leaked) then malicious plugin won't be able to decrypt it (because it was initially encrypted not by the "drainer"). So it is basically a 2 step phishing:
1. Convince a partner to encrypt his wallet private key with a malicious `X25519_PUBLIC_KEY`
2. Convince a partner to use a malicious plugin

Reply:
> Bits of entropy: log (base 2) 1000000000 = 29.9
>> I don't understand the log function being used here.

log<sub>2</sub>(1000000000)

> This I don't understand for reasons:
> 1. You're excluding the prefix 010:Repository
> 2. How many bits of entropy is required?
> 3. Isn't the string representation of the number, as it is here concatenation to the prefix, 9 bytes?

> You're excluding the prefix 010:Repository

We're excluding it because we don't need to brute force this part since (I suppose) that part is a constant for github repositories.

> How many bits of entropy is required?

- 128 bits: equivalent of 12 words ethereum mnemonic phrase
- 256 bits: equivalent of 24 words ethereum mnemonic phrase 

> Isn't the string representation of the number, as it is here concatenation to the prefix, 9 bytes?

I don't fully understand the question.

Reply:
The last part I realize doesn't matter because the attacker would be forced to use the string representation of the number anyways. 

If you are curious what I was asking about:

https://chatgpt.com/share/9bad1cf1-d14d-4bc3-841d-105fffb94b7a

Reply:
It's a bit difficult to get a grasp on the final conclusion looking at the code from my phone. What is your solution for the salt?